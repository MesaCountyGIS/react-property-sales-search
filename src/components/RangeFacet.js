import React, { Component } from 'react';
import RangeFacetLabel from './RangeFacetLabel';
import RangeBlock from './RangeBlock';
import PropTypes from 'prop-types';

const rangeFacetStyle = {
    display: 'block',
    border: '1px solid #ccc',
    borderRadius: '4px',
    padding: '0.5em 0 2em 1.5em',
    marginBottom:'1em',
    top: '0',
    height: '2.8em',
    width: '90%',
    boxShadow: '1px 1px 5px #888888'
}

  class RangeFacet extends Component {
      constructor(props){
          super(props)
      }

      changeMin = (event) =>{
          this.props.onMinChange(event.target.value);
      }
      changeMax = (event) =>{
          this.props.onMaxChange(event.target.value);
      }

    render() {
      return (
          <div style={rangeFacetStyle}>

              <RangeFacetLabel value={this.props.title} />

              <RangeBlock
                  labelText={this.props.minLabel}
                  className="facet access"
                  type="number"
                  step={this.props.step}
                  rangeValue={this.props.minValue}
                  onChange={this.changMin}
              />
              <RangeBlock
                  labelText={this.props.maxLabel}
                  className="facet access"
                  type="number"
                  step={this.props.step}
                  rangeValue={this.props.maxValue}
                  onChange={this.changMax}
              />
          </div>
      );
    }
  }

  RangeFacet.propTypes = {
      title: PropTypes.string.isRequired,
      minLabel: PropTypes.string,
      maxLabel: PropTypes.string,
      minValue: PropTypes.string.isRequired,
      maxValue: PropTypes.string.isRequired,
      step: PropTypes.string
  }

  RangeFacet.defaultProps = {
      minLabel: 'Min',
      maxLabel: 'Max',
      step: '1'
  }


export default RangeFacet;
