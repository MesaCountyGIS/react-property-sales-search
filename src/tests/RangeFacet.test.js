import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import sinon from 'sinon';
import {stubConsoleError} from './utils/test-utils'
import RangeFacet from '../components/RangeFacet';

describe("<RangeFacet />", ()=>{
    stubConsoleError();

    test(`changeMin and changeMax should update state`, ()=>{
        const wrapper = mount(
            <RangeFacet
                title='test range facet'
                minLabel='test min label'
                maxLabel='test max label'
                minValue='test min value'
                maxValue='test max value'
                step='test step'
            />
        )
        wrapper.instance().changeMin({target:{value: 500}});
        wrapper.instance().changeMax({target:{value: 5000}});
        expect(wrapper.nodes[0].state.minValue).toEqual(500);
        expect(wrapper.nodes[0].state.maxValue).toEqual(5000);
    });



            test(`renders without crashing`, () => {
                const tree = renderer.create(
                    <RangeFacet
                        title='test range facet'
                        minLabel='test min label'
                        maxLabel='test max label'
                        minValue='test min value'
                        maxValue='test max value'
                        step='test step'
                    />
      ).toJSON();

      sinon.assert.notCalled(console.error)
      expect(tree).toMatchSnapshot();
    });

    test(`ensures an error is called when required prop not passed`, () => {
      const tree = renderer.create(
          <RangeFacet
              //   title='test range facet'
              minLabel='test min label'
              maxLabel='test max label'
              //   minValue='test min value'
              //   maxValue='test max value'
              step='test step'
          />
      ).toJSON();
      sinon.assert.calledThrice(console.error)
    });

    test(`ensures an error is called when required prop not passed`, () => {
      const tree = renderer.create(
          <RangeFacet
              title='test range facet'
              minLabel='test min label'
              maxLabel='test max label'
              minValue='test min value'
              maxValue='test max value'
              step='test step'
          />
      ).toJSON();

    });

});
