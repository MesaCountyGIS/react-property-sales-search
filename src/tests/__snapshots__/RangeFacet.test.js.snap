// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RangeFacet /> renders without crashing 1`] = `
<div
  style={
    Object {
      "border": "1px solid #ccc",
      "borderRadius": "4px",
      "boxShadow": "1px 1px 5px #888888",
      "display": "block",
      "height": "2.8em",
      "marginBottom": "1em",
      "padding": "0.5em 0 2em 1.5em",
      "top": "0",
      "width": "16em",
    }
  }
>
  <label
    style={
      Object {
        "borderBottom": "1px solid #ccc",
        "color": "#C4C4C4",
        "display": "block",
        "fontSize": "0.8em",
        "fontWeight": "bold",
        "margin": "0 0 0.4em 0",
        "textAlign": "left",
      }
    }
  >
    test range facet
  </label>
  <div
    style={
      Object {
        "float": "left",
        "fontSize": "0.8em",
        "lineHeight": "1.1em",
        "paddingRight": "1em",
        "position": "relative",
      }
    }
  >
    <label
      className="range"
      style={
        Object {
          "display": "block",
        }
      }
    >
      test min label
    </label>
    <input
      className="facet access"
      onChange={[Function]}
      step="test step"
      style={
        Object {
          "width": "8em",
        }
      }
      type="number"
      value="test min value"
    />
  </div>
  <div
    style={
      Object {
        "float": "left",
        "fontSize": "0.8em",
        "lineHeight": "1.1em",
        "paddingRight": "1em",
        "position": "relative",
      }
    }
  >
    <label
      className="range"
      style={
        Object {
          "display": "block",
        }
      }
    >
      test max label
    </label>
    <input
      className="facet access"
      onChange={[Function]}
      step="test step"
      style={
        Object {
          "width": "8em",
        }
      }
      type="number"
      value="test max value"
    />
  </div>
</div>
`;
